modules/ui/UiOvertureInspector.js

```diff
@@ -0,0 +1,302 @@
+    const short = (color.length < 6);
+    const r = parseInt(short ? color[1] + color[1] : color[1] + color[2], 16);
+    const g = parseInt(short ? color[2] + color[2] : color[3] + color[4], 16);
+    const b = parseInt(short ? color[3] + color[3] : color[5] + color[6], 16);
+    return ((r * 299) + (g * 587) + (b * 114)) / 1000;
+  }
+
+
+  /**
+   * renderFeatureInfo
+   * Renders the 'feature-info' section (the dataset name)
+   * @param {d3-selection} $selection - A d3-selection to a HTMLElement that this content should render itself into
+   */
+  renderFeatureInfo($selection) {
+    const datum = this.datum;
+    if (!datum) return;
+
+    const context = this.context;
+    const l10n = context.systems.l10n;
+    const rapid = context.systems.rapid;
+
+    const datasetID = datum.__datasetid__;
+    const dataset = rapid.datasets.get(datasetID);
+    const color = dataset.color;
+
+    let $featureInfo = $selection.selectAll('.feature-info')
+      .data([0]);
+
+    // enter
+    const $$featureInfo = $featureInfo.enter()
+      .append('div')
+      .attr('class', 'feature-info');
+
+    $$featureInfo
+      .append('div')
+      .attr('class', 'dataset-label');
+
+    // update
+    $featureInfo = $featureInfo.merge($$featureInfo);
+
+    $featureInfo
+      .style('background', color)
+      .style('color', this.getBrightness(color) > 140.5 ? '#333' : '#fff');
+
+    // Attempt to localize the dataset name, fallback to 'label' or 'id'
+    const text = dataset.labelStringID ? l10n.t(dataset.labelStringID) : (dataset.label || dataset.id);
+    $featureInfo.selectAll('.dataset-label')
+      .text(text);
+  }
+
+
+  /**
+   * renderPropertyInfo
+   * Renders the 'property-info' section
+   * @param {d3-selection} $selection - A d3-selection to a HTMLElement that this content should render itself into
+   */
+  renderPropertyInfo($selection) {
+    const properties = this.datum?.geojson.properties;
+    if (!properties) return;
+
+    const context = this.context;
+    const l10n = context.systems.l10n;
+
+    let $propInfo = $selection.selectAll('.property-info')
+      .data([0]);
+
+    // enter
+    const $$propInfo = $propInfo.enter()
+      .append('div')
+      .attr('class', 'property-info');
+
+    const $$propBag = $$propInfo
+      .append('div')
+      .attr('class', 'property-bag');
+
+
+      //Overture properties can come to us as strings, JSON arrays, or JSON objects. Handle all three!
+    for (const [k, v] of Object.entries(properties)) {
+      const $$propHeading = $$propBag
+      .append('div')
+      .attr('class', 'property-heading');
+
+      let key = k;
+
+      // Some params come to us via pmtiles with a prepended '@' sign.
+      if (key.startsWith('@')) {
+        key = key.slice(1);
+      }
+      key = key.charAt(0).toUpperCase() + key.slice(1);
+      $$propHeading.text(key);
+
+      const $$tagEntry = $$propBag.append('div').attr('class', 'property-entry');
+
+      if (this._hasJsonStructure(v)) {
```

Comment:
This is a nit, but the way it's written we are double-parsing everything - once to determine if `_hasJsonStructure` is true/false, then again to actually get the json.  It would be more efficient if `_hasJsonStructure` was instead `_getJson` and it would always return the parsed json or maybe `{}` if it failed.

Reply:
Yeah JSON can be kind of weird.  Even `null` and `'null'` are "valid" JSON - you can do `JSON.parse('null')` and it will work (i.e. not throw).   

So attempting to look at it with regex is not a great solution.  It's probably better to just attempt to parse it in the `try` block, fallback to an empty result, and then deal with that result.

Reply:
Okay! Refactored into a single method and cleaned things up a little. We now handle the case where not-a-string is passed into the rendering method (we continue) and otherwise, the rendering code is the same.